The filter tables become full of stale/bad data that gets hidden.

Tables, such as `named_search_filter_group` and `filter_criterion` end up with stale data for "non-saved" filters.
There may be more tables than just those (follow table references/keys).

What is going on is that when a filter is added but is not yet "saved" an entry is created.
These are never accessible because they do not appear in the "saved filter" by design.
These also do not have names because they are not "saved".
Note that some of the UI has end points processing based on the `name` and so these might not be fetchable by such end points even if desired.

Every time either the "clear filters" or a filter is directly removed, the existing "not saved" filter is cleared and a new one is created. The data for this "not saved" filter remains in the database wasting space and is never used again. I have created a proof of concept fix for just the clear filter part in commit https://github.com/TexasDigitalLibrary/Vireo/commit/e39ebd22a48f85524edc2ddc6c0a4e6ff43bd9b8.

Preserving this stale and no longer used data almost certainly has a performance cost and may likely be a significant contributing factor to performance problems regarding the filter.

The confusion in the terminology (referenced in issue #1737 doesn't help any while investigating this issue).

### Additional Parts of the Problem
Because the data is not being deleted, saving a filter with some name such as "test" and then deleting it means that a person can never use a filter again with the name "test".

With the pre-existing behavior, the save silently fails when there is already a "test" (despite that test having been "deleted" previously).

More details on this in the comment below.

There may be additional requirements needed to be gathered. Such as, what is the expected result of removing a public filter? how to handle duplicate names of public filters? and what feedback should be presented in the UI?

To reproduce

1. Log in as admin
2. Navigate to the list view
4. Customize filters adding Title *will appear in Further Filter By sidebar box
5. Filter title by text "test"
6. Save current filter with a specific *name*
7. Remove existing filter removing the filter with the specific *name* from step 6
8. Save current filter with same specific *name* from step 6 and 8

Expected to be able to recreate filter with specified *name*.

If while saving current filter the public checkbox is not checked the following error occurs.

<img width="834" alt="image" src="https://user-images.githubusercontent.com/8352733/234111082-f364e302-0b86-4e53-a960-82e2a93e0830.png">

```
Caused by: org.postgresql.util.PSQLException: ERROR: duplicate key value violates unique constraint "uka8mgybhu9ik92wg8u3mrnytpl"
  Detail: Key (user_id, name)=(32, Test 2) already exists.
```

--- a/src/main/java/org/tdl/vireo/controller/SubmissionListController.java
+++ b/src/main/java/org/tdl/vireo/controller/SubmissionListController.java
@@ -7,8 +7,12 @@ import static org.springframework.web.bind.annotation.RequestMethod.POST;
 import edu.tamu.weaver.auth.annotation.WeaverUser;
 import edu.tamu.weaver.response.ApiResponse;
 import edu.tamu.weaver.validation.aspect.annotation.WeaverValidatedModel;
+import java.util.ArrayList;
+import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
@@ -150,16 +154,17 @@ public class SubmissionListController {
     @RequestMapping(value = "/set-active-filter", method = POST)
     public ApiResponse setActiveFilter(@WeaverUser User user, @RequestBody NamedSearchFilterGroup namedSearchFilterGroup) {
 
-        NamedSearchFilterGroup activeFilter = user.getActiveFilter();
+        Optional<NamedSearchFilterGroup> desiredFilter = namedSearchFilterGroupRepo.findById(namedSearchFilterGroup.getId());
 
-        activeFilter.getNamedSearchFilters().clear();
-        activeFilter.getSavedColumns().clear();
+        if (desiredFilter.isEmpty()) {
+            return new ApiResponse(ERROR, "Failed to find filter with ID " + namedSearchFilterGroup.getId() + ".");
+        }
 
-        activeFilter = namedSearchFilterGroupRepo.clone(activeFilter, namedSearchFilterGroup);
+        user.setActiveFilter(desiredFilter.get());
 
         user = userRepo.save(user);
 
-        simpMessagingTemplate.convertAndSend("/channel/active-filters/" + activeFilter.getId(), new ApiResponse(SUCCESS, user.getActiveFilter()));
+        simpMessagingTemplate.convertAndSend("/channel/active-filters/" + user.getActiveFilter().getId(), new ApiResponse(SUCCESS, user.getActiveFilter()));
 
         return new ApiResponse(SUCCESS);
     }
@@ -269,13 +274,50 @@ public class SubmissionListController {
     @PreAuthorize("hasRole('REVIEWER')")
     public ApiResponse clearFilterCriteria(@WeaverUser User user) {
         NamedSearchFilterGroup activeFilter = user.getActiveFilter();
+
+        // Switch to a not "saved" filter (title is NULL) when clearing.
+        List<NamedSearchFilterGroup> filters = namedSearchFilterGroupRepo.findByUserAndNameIsNull(user);
+        if (activeFilter == null || activeFilter.getName() != null) {
+            if (filters.size() == 0) {
+                activeFilter = new NamedSearchFilterGroup();
+                activeFilter.setUser(user);
+            } else {
+                activeFilter = filters.get(0);
+            }
+        }
+
+        // When clearing a not saved filter (title is NULL), then remove the existing filters and columns.
+        List<Long> searchFilterIds = new ArrayList<>();
+        List<Long> savedColumnIds = new ArrayList<>();
+
+        if (activeFilter.getName() == null) {
+            for (NamedSearchFilter searchFilter : activeFilter.getNamedSearchFilters()) {
+                searchFilterIds.add(searchFilter.getId());
+            }
+
+            for (SubmissionListColumn savedColumn : activeFilter.getSavedColumns()) {
+                savedColumnIds.add(savedColumn.getId());
+            }
+        }
+
         activeFilter.getNamedSearchFilters().clear();
         activeFilter.getSavedColumns().clear();
         activeFilter.setColumnsFlag(false);
 
+        activeFilter = namedSearchFilterGroupRepo.save(activeFilter);
+
+        user.setActiveFilter(activeFilter);
         user = userRepo.save(user);
 
-        simpMessagingTemplate.convertAndSend("/channel/active-filters/" + activeFilter.getId(), new ApiResponse(SUCCESS, user.getActiveFilter()));
+        searchFilterIds.forEach(id -> {
+            namedSearchFilterRepo.deleteById(id);
+        });
+
+        savedColumnIds.forEach(id -> {
+            submissionListColumnRepo.deleteById(id);
+        });
+
+        simpMessagingTemplate.convertAndSend("/channel/active-filters/" + user.getActiveFilter().getId(), new ApiResponse(SUCCESS, user.getActiveFilter()));
 
         return new ApiResponse(SUCCESS);
     }

--- a/src/main/java/org/tdl/vireo/model/repo/NamedSearchFilterGroupRepo.java
+++ b/src/main/java/org/tdl/vireo/model/repo/NamedSearchFilterGroupRepo.java
@@ -12,6 +12,8 @@ public interface NamedSearchFilterGroupRepo extends WeaverRepo<NamedSearchFilter
 
     public List<NamedSearchFilterGroup> findByUserIsNotAndPublicFlagTrue(User user);
 
+    public List<NamedSearchFilterGroup> findByUserAndNameIsNull(User user);
+
     public NamedSearchFilterGroup findByNameAndPublicFlagTrue(String name);
 
 }

